### PR TITLE
Render hook

### PR DESCRIPTION
**Pull request by GrandSchtroumpf**
Add a hook `render` instead of `activate` & `create` in `Plugin`.

### CODE REVIEW

File: src/engine/api-factory.ts [hunk 1]
```diff
@@ -18,6 +19,7 @@ export abstract class ApiFactory<T extends Api = any> {
       events: this.events,
       name: this.profile.name,
       profile: this.profile,
+      render: this.render ? () => this.render() : undefined,
```
Comment (yann300) [inline src/engine/api-factory.ts]: i m not quite sure about the syntax here... is the `return` keyword  missing here?

Comment (GrandSchtroumpf) [reply src/engine/api-factory.ts]: `render: this.render ? () => this.render() : undefined,` is the same as : 
```javascript
render: () => {
  if (!!this.render) {
    return () => { return this.render() }
  } else {
    return undefined
  }
}
```

File: src/engine/plugin.ts [hunk 2]
```diff
@@ -147,57 +137,33 @@ export class Plugin<T extends Api> implements PluginApi<T> {
...
-        this.postMessage({
-          action: 'request',
-          name: this.name,
-          key: 'handshake',
-        })
-      }
-    } catch (err) {
-      console.log(err)
+  public render() {
+    if (this.iframe) {
```
Comment (yann300) [inline src/engine/plugin.ts]: that might be a problem if one activate / deactivate / activate a plugin, cause the instance seems not reset upon deactivation. Perhaps should we keep a `deactivation` hook for reset this object

Comment (yann300) [reply src/engine/plugin.ts]: ah i see `deactivate`, should the iframe ref should be removed to

Comment (GrandSchtroumpf) [reply src/engine/plugin.ts]: Yes, indeed. Thanks

**Pull request merged**